### ISSUE TITLE
Crashes

### ISSUE THREAD

**Issue by XXLRay** [XXLRay/libreshot#2]
Openshot often crashes when the preview changes due to clicking on to the timeline or using the resize tool especially in big projects.
I guess this comes from a segmentation fault in mlt. The total crash can probably be avoided by placing a  try-except block around the call at the correct place. So far I haven't found out where this place is though ...

**Comment by XXLRay** [comment]
Switching the order of loading end and start time did the job

**Comment by marianomarini** [comment]
If you can reproduce the crash starting from command line, you will be able to see where it happen.
At least last module. Then send to me the list of message printed in the command line console and I will check it.

**Comment by XXLRay** [comment]
This time I actually had the following when clicking into the timeline but it's very unusual:

```
...
state saved
on_mnuRemoveClip_activate clicked
project state modified
state saved
on_mnuShiftClips_activate clicked
project state modified
state saved
project saved! - Cinerella_Slideshow
GenerateXML for an Effect
GenerateXML for an Effect
[mp3 @ 0x6fbbc280] max_analyze_duration reached
[mp3 @ 0x6fbbc280] Estimating duration from bitrate, this may be inaccurate
[mp3 @ 0x6fbbe360] max_analyze_duration reached
[mp3 @ 0x6fbbe360] Estimating duration from bitrate, this may be inaccurate
[mp3 @ 0x6fbf9200] max_analyze_duration reached
[mp3 @ 0x6fc09bc0] max_analyze_duration reached
[mp3 @ 0x715051e0] max_analyze_duration reached
[mp3 @ 0x7150d300] max_analyze_duration reached
python: malloc.c:3700: _int_malloc: Assertion `victim->bk_nextsize->fd_nextsize == victim' failed.
Aborted (core dumped)
```

Usually it ends with "Segmentation Fault"

**Comment by XXLRay** [comment]
Another atypical message when clicking the timeline for preview:

```
...
GenerateXML for an Effect
GenerateXML for an Effect
[mp3 @ 0x86411ae0] max_analyze_duration reached
[mp3 @ 0x86411ae0] Estimating duration from bitrate, this may be inaccurate
[mp3 @ 0x86423be0] max_analyze_duration reached
[mp3 @ 0x86423be0] Estimating duration from bitrate, this may be inaccurate
[mp3 @ 0x876617e0] max_analyze_duration reached
[mp3 @ 0x8766df20] max_analyze_duration reached
[mp3 @ 0x88259b80] max_analyze_duration reached
[mp3 @ 0x882773c0] max_analyze_duration reached
Killed
```

**Comment by XXLRay** [comment]
Here is a typical one

```
...
on_tlbPlay_clicked called with self.GtkToolButton
on_tlbPlay_clicked called with self.GtkToolButton
on_mnuShiftClips_activate clicked
project state modified
state saved
on_mnuShiftClips_activate clicked
project state modified
state saved
project saved! - Tools_Hanginghatchet
[mp3 @ 0x191e6a60] max_analyze_duration reached
[mp3 @ 0x191e7400] max_analyze_duration reached
[mp3 @ 0x18649620] max_analyze_duration reached
[mp3 @ 0x18649fc0] max_analyze_duration reached
[mp3 @ 0x16f7a100] max_analyze_duration reached
[mp3 @ 0x1148ada0] max_analyze_duration reached
[mp3 @ 0x18630b60] max_analyze_duration reached
[mp3 @ 0xe71d000] max_analyze_duration reached
Segmentation fault (core dumped)
```

**Comment by wydevices** [comment]
Hace you observed if it's more stable if you mute all audio?

I can see a pattern involving mp3. Normally, mp3 layout is difficult to deal with it, garbage, id3.... Therefore i would say to focus on audio rather than video on this particular issue. What do you think?

-----Mensaje original-----
De: "XXLRay" notifications@github.com
Enviado: ‎13/‎10/‎2015 14:13
Para: "XXLRay/libreshot" libreshot@noreply.github.com
Asunto: Re: [libreshot] Crashes (#2)

Here is a typical one
...
on_tlbPlay_clicked called with self.GtkToolButton
on_tlbPlay_clicked called with self.GtkToolButton
on_mnuShiftClips_activate clicked
project state modified
state saved
on_mnuShiftClips_activate clicked
project state modified
state saved
project saved! - Tools_Hanginghatchet
[mp3 @ 0x191e6a60] max_analyze_duration reached
[mp3 @ 0x191e7400] max_analyze_duration reached
[mp3 @ 0x18649620] max_analyze_duration reached
[mp3 @ 0x18649fc0] max_analyze_duration reached
[mp3 @ 0x16f7a100] max_analyze_duration reached
[mp3 @ 0x1148ada0] max_analyze_duration reached
[mp3 @ 0x18630b60] max_analyze_duration reached
[mp3 @ 0xe71d000] max_analyze_duration reached
Segmentation fault (core dumped)

**Comment by XXLRay** [comment]
Haven't tried it yet but seems to be a good idea anyway. Nevertheless something should catch the exception instead of crashing the whole program.

**Comment by wydevices** [comment]
Yeah! Qualifying the problem through essay-error/success should help to identify where we need to add that error control mechanism.... Hopefully!

If the underlying problem is really an audio issue, we can focus on it :-)

-----Mensaje original-----
De: "XXLRay" notifications@github.com
Enviado: ‎13/‎10/‎2015 19:02
Para: "XXLRay/libreshot" libreshot@noreply.github.com
CC: "Wydevices" roberto.clemente.bustamante@gmail.com
Asunto: Re: [libreshot] Crashes (#2)

Haven't tried it yet but seems to be a good idea anyway. Nevertheless something should catch the exception instead of crashing the whole program.

**Comment by wydevices** [comment]
Now that we can try debugging with an IDE... this might be easier?

**Comment by XXLRay** [comment]
Oh - didn't have the time to take care of it since quite some time

**Comment by wydevices** [comment]
:-)

-----Mensaje original-----
De: "XXLRay" notifications@github.com
Enviado: ‎10/‎06/‎2016 13:56
Para: "XXLRay/libreshot" libreshot@noreply.github.com
CC: "Wydevices" roberto.clemente.bustamante@gmail.com; "Comment" comment@noreply.github.com
Asunto: Re: [XXLRay/libreshot] Crashes (#2)

Oh - didn't have the time to take care of it since quite some time

**Comment by rolandixor** [comment]
So are you planning to revive this? Because we really need someone to get an active fork of openshot going. Obviously, the developer has quit and doesn't want to admit it.

**Comment by marianomarini** [comment]
I see the forum drop down since release 2.0.
I've not much time spending on 2.0 tests, so I assume that all is going on very well.
Why we need keep going the LibreShot fork?

**Comment by wydevices** [comment]
Don't expect evolutions... Just bug fixing... If any!  No spare time enough!

-----Mensaje original-----
De: "marianomarini" notifications@github.com
Enviado: ‎14/‎06/‎2016 9:44
Para: "XXLRay/libreshot" libreshot@noreply.github.com
CC: "Wydevices" roberto.clemente.bustamante@gmail.com; "Comment" comment@noreply.github.com
Asunto: Re: [XXLRay/libreshot] Crashes (#2)

I see the forum drop down since release 2.0.
I've not much time spending on 2.0 tests, so I assume that all is going on very well.
Why we need keep going the LibreShot fork?

**Comment by XXLRay** [comment]
It would to speed up the bug fixing process if you provided some tests yourself.
